src/app/services/views/ServiceDetailView.vue

```diff
@@ -0,0 +1,94 @@
+<template>
+  <div class="service-details">
+    <LoadingBlock v-if="isLoading" />
+
+    <ErrorBlock
+      v-else-if="error !== null"
+      :error="error"
+    />
+
+    <EmptyBlock v-else-if="service === null" />
+
+    <ServiceDetails
+      v-else
+      :service="service"
+      :data-plane-overviews="dataPlaneOverviews"
+      :external-service="externalService"
+      @load-data="loadData"
+    />
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { ref, watch } from 'vue'
+import { useRoute } from 'vue-router'
+
+import { DataPlaneOverview, ExternalService, ServiceInsight } from '@/types/index.d'
+import { kumaApi } from '@/api/kumaApi'
+import { useStore } from '@/store/store'
+import EmptyBlock from '@/app/common/EmptyBlock.vue'
+import ErrorBlock from '@/app/common/ErrorBlock.vue'
+import LoadingBlock from '@/app/common/LoadingBlock.vue'
+import ServiceDetails from '../components/ServiceDetails.vue'
+
+const route = useRoute()
+const store = useStore()
+
+const service = ref<ServiceInsight | null>(null)
+const externalService = ref<ExternalService | null>(null)
+const dataPlaneOverviews = ref<DataPlaneOverview[] | null>(null)
+const isLoading = ref(true)
+const error = ref<Error | null>(null)
+
+watch(() => route.params.mesh, function () {
+  // Don’t trigger a load when the user is navigating to another route.
+  if (route.name !== 'service-insights-detail-view') {
+    return
+  }
+
+  loadData()
+})
+
+watch(() => route.params.name, function () {
+  // Don’t trigger a load when the user is navigating to another route.
+  if (route.name !== 'service-insights-detail-view') {
+    return
+  }
```

Comment:
Bit of a general question: Is this a typical Vue thing that we need to remember to add these guards all the time in `watch`es?